src/AST-Core/RBDumpVisitor.class.st

```diff
@@ -0,0 +1,214 @@
+"
+I represent a visitor that generates Pharo code
```

Comment:
I'm a visitor that generates code whose execution will recreate the visited node (similarly to storeOn: protocol). 
This is handy because we can simply serialize the object in a textual form without requiring a separate parser. 

```
(RBParser parseExpression: (Point >> #asNonFractionalPoint) ast dump ) formattedCode 

'RBMethodNode
	selector: #asNonFractionalPoint
	body:
		(RBSequenceNode
			statements:
				{(RBMessageNode
					receiver:
						(RBMessageNode
							receiver:
								(RBMessageNode
									receiver: (RBInstanceVariableNode named: #x)
									selector: #isFraction)
							selector: #or:
							arguments:
								{(RBBlockNode
									body:
										(RBSequenceNode
											statements:
												{(RBMessageNode
													receiver: (RBInstanceVariableNode named: #y)
													selector: #isFraction)}))})
					selector: #ifTrue:
					arguments:
						{(RBBlockNode
							body:
								(RBSequenceNode
									statements:
										{(RBReturnNode
											value:
												(RBMessageNode
													receiver:
														(RBMessageNode
															receiver: (RBInstanceVariableNode named: #x)
															selector: #asFloat)
													selector: #@
													arguments:
														{(RBMessageNode
															receiver: (RBInstanceVariableNode named: #y)
															selector: #asFloat)}))}))})})'
```

Reply:
Thanks, that's indeed better than what I wrote !